Model/Sync/Category/Processor/ProcessByCategory.php

```diff
@@ -140,30 +140,24 @@ public function processEntity($retryCategoryIds = [])
         $currentTime = date('Y-m-d H:i:s');
         $batchSize = $this->config->getConfig('product_sync_limit');
         $existColls = [];
-        $attributeId = $this->data->getAttributeId(Config::CATEGORY_SYNC_ATTR_CODE);
-        $collection = $this->categoryCollectionFactory->create();
-        $collection->addAttributeToSelect('*');
-        $collection->addNameToResult();
+        $collection = $this->categoryHelper->getStoreCategories(true, true);
         if ($retryCategoryIds) {
-            $collection->getSelect()
-                ->where('e.entity_id in (?)', $retryCategoryIds);
+            /** @phpstan-ignore-next-line */
+            $collection->addFieldToFilter('entity_id', ['in' => $retryCategoryIds]);
         }
-        $collection->getSelect()->joinLeft(
-            ['at' => $this->resourceConnection->getTableName('catalog_category_entity_int')],
-            'e.' . $this->entityIdFieldValue . ' = at.' . $this->entityIdFieldValue .
-            ' AND at.attribute_id = ' . $attributeId .
-            ' AND at.store_id=\'' . $this->config->getStoreId() . '\'',
-            null
-        );
         if (!$retryCategoryIds) {
```

Comment:
@manaftryzens Why you're not using `else` clause?

```suggestion
        if ($retryCategoryIds) {
           $collection->addFieldToFilter('entity_id', ['in' => $retryCategoryIds]);
        } else {
                       $collection->addAttributeToFilter(
                 [
                     ['attribute' => Config::CATEGORY_SYNC_ATTR_CODE, 'null' => true],
                     ['attribute' => Config::CATEGORY_SYNC_ATTR_CODE, 'eq' => '0'],
                 ]
             );
          }
```